Include/internal/pycore_object.h

```diff
@@ -16,7 +16,6 @@ extern "C" {
 #include "pycore_pystate.h"       // _PyInterpreterState_GET()
 #include "pycore_typeid.h"        // _PyType_IncrefSlow
 
-
```

Comment:
I wonder if in `_Py_TryIncRefShared` we could just do an atomic add to the `ob_ref_shared` if the object is already marked as weakref'd? E.g:

    Py_ssize_t shared = _Py_atomic_load_ssize_relaxed(&op->ob_ref_shared);
    for (;;) {
        if ((shared & _Py_REF_SHARED_FLAG_MASK) == _Py_REF_MAYBE_WEAKREF) {
            _Py_atomic_add_ssize(&op->ob_ref_shared, (1 << _Py_REF_SHARED_SHIFT));
           return 1;
        }
    }

That might avoid the threads spinning on updating the value when the value is in heavy contention and instead all of the atomic operations should just slowly proceed.